### ISSUE TITLE
Improve the wiki multiprocess documentation

### ISSUE THREAD

**Issue by ceyusa** [servo/servo#24221]
Right now the only reference to the multiprocess design is in https://github.com/servo/servo/wiki/Design#multi-process-architecture

It would be great if there are more design notes, or just maybe a diagram specifying the processes and their relations. Maybe aiming to code where currently it is done.

**Comment by asajeffrey** [comment]
Yeah, the docs for the process architecture keep drifting away from the impl. I wonder what we can do about this, the only way I can think of keeping them in sync is to move to having the docs in the source, but that's not ideal either.

**Comment by gterzian** [comment]
The specification of [agent-cluster](https://html.spec.whatwg.org/#integration-with-the-javascript-agent-cluster-formalism) could anchor our architecture more firmly into a spec going forward. 

Some overview is proposed at https://github.com/servo/servo/issues/24159 and I think its a good idea to update the Wiki as well.

We could also add some notes about the actual tools used, like https://github.com/servo/gaol and https://github.com/servo/ipc-channel/.
